File: BackEnd/NDGradGateKernel/4-Domain/ND.GradGate.Kernel.Domain/Faculty/Faculty.cs
using ND.GradGate.Kernel.Domain.Core.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace ND.GradGate.Kernel.Domain.Faculty
{
    public class Faculty : AbstractEntity<Faculty, int>
    {
        public string Email { get; set; }
        public string Name { get; set; }
        public bool IsAdmin { get; set; }
        public bool IsReviewer { get; set; }
        public string Field { get; set; }

        // Relationships
        public virtual ICollection<PotentialAdvisor> PotentialAdvisors { get; set; }
        public virtual ICollection<ReviewerAssignment> ReviewerAssignments { get; set; }
        public virtual ICollection<Comment> Comments { get; set; }
    }

}

File: FrontEnd/ndgradgate/src/types/Application/Faculty.ts
export interface PotentialAdvisorDto {
  Id: number;
}

export interface ReviewerAssignmentDto {
  AssignmentId: number;
  ReviewerName: string;
  ReviewDate: Date;
}

export interface CommentDto {
  FacultyId: number;
  ApplicantId: number;
  Content: string;
}

export interface Faculty {
  Id: number;
  Name: string;
  Email: string;
  IsAdmin: boolean;
  IsReviewer: boolean;
  Field: string;
  PotentialAdvisors: PotentialAdvisorDto[];
  ReviewerAssignments: ReviewerAssignmentDto[];
  Comments: CommentDto[];
}
